### ISSUE TITLE
Massive performance discrepancy between net451 and netcoreapp1.0 with large parameterization

### ISSUE THREAD

**Issue by jskeet** [nunit/dotnet-test-nunit#109]
I've only reproduced this in .NET Core so far, in the old project.json format - I wanted to report it rather than sit on it, but I understand this repo doesn't have a future. Once support for Core on csproj is done, I'll be happy to retest there.

`project.json`:

```json
{
  "buildOptions": {
    "optimize": true
  },

  "dependencies": {
    "NUnit": "3.6.1",
    "dotnet-test-nunit": "3.4.0-beta-1"
  },

  "testRunner": "nunit",

  "frameworks": {
    "net451": { },
    "netcoreapp1.0": {
      "dependencies": {
        "Microsoft.NETCore.App": {
          "version": "1.0.0",
          "type": "platform"
        }
      }
    }
  }
}
```

And `PerfTest.cs`:

```csharp
using NUnit.Framework;
using System.Linq;

namespace NodaTime.Test.Text
{
    public class PerfTest
    {
        private static readonly TestCaseData[] Data =
            (from x in Enumerable.Range(1, 500)
             from y in Enumerable.Range(1, 10)
             select new TestCaseData(x, y)).ToArray();

        [Test]
        [TestCaseSource(nameof(Data))]
        public void Foo(int x, int y)
        {
        }
    }
}
```

Running `dotnet test -f net451` takes 20 seconds, whereas running `dotnet test -f netcoreapp1.0` takes only 2 seconds. I haven't worked out why yet.

**Comment by jskeet** [comment]
This is still a problem when using the .NET Core 2.0 SDK and NUnitLite console application. Should I refile this in a different project? (I may investigate it a bit further with a profiler...)

**Comment by jskeet** [comment]
I strongly suspect this is due to https://github.com/nunit/nunit/issues/2217

**Comment by ChrisMaddock** [comment]
@jskeet - nunit/nunit#2217 should be fixed in master. (https://github.com/nunit/nunit/pull/2233)

Can you try a nightly build from MyGet? 

https://www.myget.org/F/nunit/api/v2

**Comment by jskeet** [comment]
@ChrisMaddock: Confirmed - with 3.8.0-dev-04276 it hardly takes any time at all.

(Not sure whether this should therefore be closed, or only when 3.8.0 is released...)

**Comment by ChrisMaddock** [comment]
Great - thanks! 

Will close it now. We normally close when fixes are in master, I think nunit/nunit#2217 had just lead on to further discussion. 🙂
